sklearn/utils/testing.py

```diff
@@ -422,6 +441,172 @@ def assert_raise_message(exceptions, message, function, *args, **kwargs):
+
+    If the models are similar an AssertionError with the given error message
+    is raised.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        Input data, for the fitted models, used for comparing them.
+
+    estimator1 : An estimator object.
+        The first fitted model to be compared.
+
+    estimator2 : An estimator object.
+        The second fitted model to be compared.
+
+    msg : str
+        The error message to be used while raising the AssertionError if the
+        models are similar.
+
+    Notes
+    -----
+    This check is not exhaustive since all attributes of the model are assumed
+    to end with ``_``. If that is not the case, it could lead to false
+    negatives.
+    """
+    try:
+        assert_same_model(X, estimator1, estimator2)
+    except AssertionError:
+        pass
+    else:
+        raise AssertionError(msg)
+
+
+def assert_fitted_attributes_equal(estimator1, estimator2, msg=None):
+    """Helper function to check if the fitted model attributes are similar.
+
+    This check is done by comparing the attributes from both the models that
+    end in ``_``.
+
+    If the fitted models attributes are dissimilar an AssertionError with the
+    given error message is raised.
+
+    Parameters
+    ----------
+    estimator1 : An estimator object.
+        The first fitted model whose attributes are to be compared.
+
+    estimator2 : An estimator object.
+        The second fitted model whose attributes are to be compared.
+
+    msg : str
+        The error message to be used while raising the AssertionError, if the
+        fitted models attributes are dissimilar.
+
+    Notes
+    -----
+    This check is not exhaustive since all attributes of the model are assumed
+    to end with ``_``. If that is not the case, it could lead to false
+    positives.
+    """
+    est1_dict = estimator1.__dict__
+    est2_dict = estimator2.__dict__
+
+    assert_equal(est1_dict.keys(), est2_dict.keys(),
+                 "The attributes of both the estimators do not match.")
+
+    for attr in est1_dict:
+        # Consider keys that end in ``_`` only as attributes.
+        if (attr.endswith('_') and (attr.endswith('estimator_') or
+                                    attr.endswith('estimators_'))):
+
+            msg = ("The attribute, %s, does not match. \nThe attribute, %s, in"
+                   " estimator1,\n\n%r\n\n is %r and in estimator2, \n\n%r\n\n"
+                   " is %r.\n") % (attr, attr, estimator1, est1_dict[attr],
+                                   estimator2, est2_dict[attr])
+
+            if sp.issparse(est1_dict[attr]) and sp.issparse(est2_dict[attr]):
+                assert_equal(est1_dict[attr].shape, est2_dict[attr].shape, msg)
+                _assert_sparse_allclose(est1_dict[attr], est2_dict[attr],
+                                        err_msg=msg)
+            else:
+                assert_equal(est1_dict[attr], est2_dict[attr], msg)
```

Comment:
This seems wrong to me, because the `assert_equal` requires strict equality. Is this the intended behavior?